src/pydocstyle/violations.py

```diff
@@ -29,10 +38,10 @@ def __init__(self, code, short_desc, context, *parameters):
         self.short_desc = short_desc
         self.context = context
         self.parameters = parameters
-        self.definition = None
-        self.explanation = None
+        self.definition = None  # type: Optional[Definition]
+        self.explanation = None  # type: Optional[str]
```

Comment:
Can't we use the `self.explanation: Optional[str] = None` annotation?

Reply:
Non-comment variable type hints were introduced only in Python 3.6 (with [PEP 526](https://www.python.org/dev/peps/pep-0526/)).